block/block.md

```diff
@@ -1 +1,99 @@
 # Block
+
+## Abstract
+
+The block manager is a key component of full nodes that is responsible for block production or block syncing depending on the node type. Block syncing in this context includes retrieving the published blocks from the network, validating them to raise fraud proofs upon validation failure, and updating the state and storing the validated blocks. A full node invokes multiple of the block manager functionalities in parallel, such as:
+
+* block production (only for sequencer full nodes)
+* block publication to DA network
+* block retrieval from DA network
+* block retrieval from Blockstore (which retrieves blocks from the p2p network)
+* block syncing
+
+## Protocol/Component Description
+
+The block manager is initialized using several parameters as defined below:
+
+* `signing key`: used for signing the blocks after production
+* `config`: block manager configurations (see config options below)
+* `genesis`: initialize the block manager with genesis state
+* `localstore`: store blocks and states
+* `(mempool, proxyapp, eventbus)`: for initializing the executor (state transition function). mempool is also used in the manager to check for availability of transactions for lazy block production
+* `dalc`: the data availability light client used to submit and retrieve blocks to DA network  
+* `blockstore`: to retrieve blocks from block store (p2p blocksync)
+
+Block manager configuration options:
+
+* `BlockTime`: time interval used for block production and block retrieval from Blockstore (default: 1 second)
```

Comment:
We should be consistent with how we use the word "block store" or "Blocktore" everywhere
```suggestion
* `BlockTime`: time interval used for block production and block retrieval from block store (default: 1 second)
```